File: src/main/java/programming_scala/scalaForTheJavaEyes/Car.scala
// 主构造函数
class Car(val year: Int) {
  private var milesDriven: Int = 0
  def miles() = milesDriven
  def drive(distance: Int): Unit = {
    milesDriven += distance
  }
}

val car = new Car(2009)
println(car.year)
println(car.miles)
car.drive(100)
car.drive(200)
println(car.miles)

File: src/main/resources/programming_scala/scalaForTheJavaEyes/Car.java
package programming_scala.scalaForTheJavaEyes;

public class Car {
    private final int year;
    private int miles;

    public Car(int yearOfMake) {
        year = yearOfMake;
    }

    public int getYear() {
        return year;
    }

    public int getMiles() {
        return miles;
    }

    public void driver(int distance) {
        this.miles += Math.abs(distance);
    }
}
